Orange/widgets/unsupervised/owmds.py

```diff
@@ -1017,336 +689,37 @@ def commit(self):
 
-        if output is not None and self._selection_mask is not None and \
-                numpy.any(self._selection_mask):
-            subset = output[self._selection_mask]
-            self.selection_indices = numpy.flatnonzero(self._selection_mask)
+        selection = self.graph.get_selection()
+        if output is not None and len(selection) > 0:
+            selected = output[selection]
         else:
-            self.selection_indices = subset = None
-        self.Outputs.selected_data.send(subset)
-        self.Outputs.annotated_data.send(create_annotated_table(output, self._selection_mask))
+            selected = None
+        self.Outputs.selected_data.send(selected)
+        self.Outputs.annotated_data.send(create_annotated_table(output, selection))
 
     def onDeleteWidget(self):
         super().onDeleteWidget()
         self._clear_plot()
         self._clear()
 
-    def __selection_end(self, path):
-        self.select(path)
-        self._pen_data = None
-        self._update_plot()
-        self._invalidate_output()
-
-    def select(self, region):
-        item = self._scatter_item
-        if item is None:
-            return
-
-        indices = numpy.array(
-            [spot.data() for spot in item.points()
-             if region.contains(spot.pos())],
-            dtype=int)
-
-        if not QApplication.keyboardModifiers():
-            self._selection_mask = None
-
-        self.select_indices(indices, QApplication.keyboardModifiers())
-
-    def select_indices(self, indices, modifiers=Qt.NoModifier):
-        if self.data is None or indices is None:
-            return
-
-        if self._selection_mask is None or \
-                not modifiers & (Qt.ControlModifier | Qt.ShiftModifier |
-                                 Qt.AltModifier):
-            self._selection_mask = numpy.zeros(len(self.data), dtype=bool)
-
-        if modifiers & Qt.AltModifier:
-            self._selection_mask[indices] = False
-        elif modifiers & Qt.ControlModifier:
-            self._selection_mask[indices] = ~self._selection_mask[indices]
-        else:
-            self._selection_mask[indices] = True
-
     def send_report(self):
         if self.data is None:
             return
-        self.report_plot()
+        def name(var):
+            return var and var.name
         caption = report.render_items_vert((
-            ("Color", self.color_value != "Same color" and self.color_value),
-            ("Shape", self.shape_value != "Same shape" and self.shape_value),
-            ("Size", self.size_value != "Same size" and self.size_value),
-            ("Labels", self.label_value != "No labels" and self.label_value)))
+            ("Color", name(self.graph.attr_color)),
+            ("Label", name(self.graph.attr_label)),
+            ("Shape", name(self.graph.attr_shape)),
+            ("Size", name(self.graph.attr_size)),
+            ("Jittering", self.graph.jitter_size)))
```

Comment:
Color, label, shape and size are not reported when set to None (which is OK). Do the same for jittering. Don't report it when set to 0; add ` %`, when non-zero. That is, do something like:

```
("Jittering", self.graph.jitter_size != 0 and "{} %".format(self.graph.jitter_size))
```

Reply:
Ok.